google_guest_agent/hostname/hostname_test.go

```diff
@@ -0,0 +1,537 @@
+					SetFqdn:        true,
+				},
+			},
+			setFqdn:     func(string, string) error { return fmt.Errorf("fqdn failure") },
+			setHostname: func(string) error { return nil },
+			req:         ReconfigureHostnameRequest{},
+			resp: ReconfigureHostnameResponse{
+				Response: command.Response{Status: 2, StatusMessage: "fqdn failure"},
+				Hostname: "host1",
+			},
+			lastHostname: "host1",
+			lastFqdn:     "host1.example.com",
+		},
+		{
+			name: "fail to reconfigure hostname and fqdn",
+			cfg: &cfg.Sections{
+				Unstable: &cfg.Unstable{
+					FqdnAsHostname: false,
+					SetHostname:    true,
+					SetFqdn:        true,
+				},
+			},
+			setFqdn:     func(string, string) error { return fmt.Errorf("fqdn failure") },
+			setHostname: func(string) error { return fmt.Errorf("hostname failure") },
+			req:         ReconfigureHostnameRequest{},
+			resp: ReconfigureHostnameResponse{
+				Response: command.Response{Status: 3, StatusMessage: "hostname failurefqdn failure"},
+			},
+			lastHostname: "host1",
+			lastFqdn:     "host1.example.com",
+		},
+		{
+			name: "empty hostname",
+			cfg: &cfg.Sections{
+				Unstable: &cfg.Unstable{
+					FqdnAsHostname: false,
+					SetHostname:    true,
+					SetFqdn:        true,
+				},
+			},
+			setFqdn:     func(string, string) error { return nil },
+			setHostname: func(string) error { return nil },
+			req:         ReconfigureHostnameRequest{},
+			resp: ReconfigureHostnameResponse{
+				Response: command.Response{Status: 1, StatusMessage: "disallowed hostname: \"\""},
+				Fqdn:     "host1.example.com",
+			},
+			lastHostname: "",
+			lastFqdn:     "host1.example.com",
+		},
+		{
+			name: "empty fqdn",
+			cfg: &cfg.Sections{
+				Unstable: &cfg.Unstable{
+					FqdnAsHostname: false,
+					SetHostname:    true,
+					SetFqdn:        true,
+				},
+			},
+			setFqdn:     func(string, string) error { return nil },
+			setHostname: func(string) error { return nil },
+			req:         ReconfigureHostnameRequest{},
+			resp: ReconfigureHostnameResponse{
+				Response: command.Response{Status: 2, StatusMessage: "disallowed fqdn: \"\""},
+				Hostname: "host1",
+			},
+			lastHostname: "host1",
+			lastFqdn:     "",
+		},
+		{
+			name: "mds name as hostname",
+			cfg: &cfg.Sections{
+				Unstable: &cfg.Unstable{
+					FqdnAsHostname: false,
+					SetHostname:    true,
+					SetFqdn:        true,
+				},
+			},
+			setFqdn:     func(string, string) error { return nil },
+			setHostname: func(string) error { return nil },
+			req:         ReconfigureHostnameRequest{},
+			resp: ReconfigureHostnameResponse{
+				Response: command.Response{Status: 3, StatusMessage: "disallowed hostname: \"metadata.google.internal\"disallowed fqdn: \"metadata.google.internal\""},
+			},
+			lastHostname: "metadata.google.internal",
+			lastFqdn:     "metadata.google.internal",
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			cfg.Get().Unstable = tc.cfg.Unstable
+			setFqdn = tc.setFqdn
```

Comment:
can we use tc directly instead of defining these variables? (I don't see it being changed anywhere)

Reply:
These variables are the internal state that `ReconfigureHostname` accesses, they're not local to the test.